code/game/objects/structures/razorwire.dm

```diff
@@ -0,0 +1,164 @@
+/obj/structure/razorwire
+	name = "razorwire"
+	desc = "A mesh of metal strips with sharp edges whose purpose is to prevent passage by humans."
+	icon = 'icons/obj/structures/razorwire.dmi'
+	icon_state = "razorwire-"
+	layer = BELOW_OBJ_LAYER
+
+	var/yautja_slowdown = 0.1 SECONDS
+	var/human_slowdown = 0.5 SECONDS
+	var/xeno_slowdown = 0.3 SECONDS
+
+	var/brute_multiplier = 1
+	var/burn_multiplier = 1
+	var/explosive_multiplier = 1
+
+	var/force_level_absorption = 5 //How much force an item needs to even damage it at all.
+
+	health = 250
+
+/obj/structure/razorwire/Initialize()
+	. = ..()
+	update_icons_in_range(include_self = TRUE)
+
+/obj/structure/razorwire/Destroy()
+	var/turf/old_turf = get_turf(src)
+	. = ..()
+	update_icons_in_range(include_self = FALSE, source = old_turf)
+
+/obj/structure/razorwire/proc/update_icons_in_range(var/include_self = TRUE, var/atom/source = src)
+	for(var/direction in cardinal)
+		var/turf/razor_turf = get_step(source, direction)
+		var/obj/structure/razorwire/razor = locate() in razor_turf
+		if(razor)
+			razor.update_icon()
+	if(include_self)
+		update_icon()
+
+/obj/structure/razorwire/examine(mob/user)
+	..()
+	var/health_fraction = health / initial(health)
```

Comment:
make a maxhealth var instead of using initial(health) because mappers like to instance partially damaged cades and defences

Reply:
then they can just adjust health, initial(health) refers to the coded instance iirc